src/MultiRunner.php

```diff
@@ -84,18 +89,27 @@ public function remove(PromiseCore $core): void
     public function wait(PromiseCore $targetCore = null): void
     {
         do {
-            $status = curl_multi_exec($this->multiHandle, $active);
+            if (curl_multi_select($this->multiHandle, self::SELECT_TIMEOUT) === -1) {
+                // See https://bugs.php.net/bug.php?id=61141
+                usleep(250);
```

Comment:
so we need a sleep because the timeout does not actually work and curl_multi_select is broken?

Reply:
https://bugs.php.net/bug.php?id=61141#1348320122:

> Internally php curl_multi_select uses libcurl curl_multi_fdset function to set 
all the fd_set and the maxfd value. The libcurl curl_multi_fdset documentation 
says :
> When libcurl returns -1 in max_fd, it is because libcurl currently does 
something that isn't possible for your application to monitor with a socket and 
unfortunately you can then not know exactly when the current action is completed 
using select(). When max_fd returns with -1, you need to wait a while and then 
proceed and call curl_multi_perform anyway. How long to wait? I would suggest 
100 milliseconds at least, but you may want to test it out in your own 
particular conditions to find a suitable value.

Reply:
ugh. guess it makes sense, and saw that guzzle is doing the same, so i guess its what we should do.